packages/core/src/apex/ApexClassFetcher.ts

```diff
@@ -11,9 +12,13 @@ export default class ApexClassFetcher {
      * @returns
      */
     public async fetchApexClassByName(classNames: string[]): Promise<{ Id: string; Name: string }[]> {
-        let collection = classNames.map((name) => `'${name}'`).toString(); // transform into formatted string for query
-        let query = `SELECT ID, Name FROM ApexClass WHERE Name IN (${collection})`;
+        const chunks = chunkCollection(classNames);
 
-        return QueryHelper.query<{ Id: string; Name: string }>(query, this.conn, false);
+        for (const chunk of chunks) {
+            const formattedChunk = chunk.map(elem => `'${elem}'`).toString(); // transform into formatted string for query
+            const query = `SELECT ID, Name FROM ApexClass WHERE Name IN (${formattedChunk})`;
+
+            return QueryHelper.query<{ Id: string; Name: string }>(query, this.conn, false);
```

Comment:
Is the logic correct here, so after first chunk in the loop, it returns?

Reply:
Yep, you're right it should be aggregated.